pkg/systemd/services/services.jsx

```diff
@@ -252,65 +280,98 @@
+        /* Start listening to signals for updates
+         * - when in the middle of reload mute all signals
          * - We don't need to listen to 'UnitFilesChanged' signal since every time we
-         *   perform some file operation we do call Reload which issues 'Reload' signal
-         * - JobNew is also useless, JobRemoved is enough since it comes in pair with JobNew
-         *   but we are interested to update the state when the operation finished
+         *   perform some file operation we do call Reload which issues 'Reloading' signal
          */
         systemd_client[this.props.owner].subscribe({
-            interface: "org.freedesktop.DBus.Properties",
+            interface: s_bus.I_PROPS,
             member: "PropertiesChanged"
-        }, (path, iface, signal, args) => {
-            if (this.props.isLoading)
+        }, async (path, _iface, _signal, [iface, props]) => {
+            if (this.props.isLoading || this.reloading)
                 return;
 
-            if (this.state.unit_by_path[path] &&
-                this.state.unit_by_path[path].Transient &&
-                args[1].ActiveState &&
-                ((args[1].ActiveState.v == 'failed' && this.state.unit_by_path[path].CollectMode == 'inactive-or-failed') ||
-                  args[1].ActiveState.v == 'inactive')) {
-                this.seenPaths.delete(path);
-                const copy_unit_by_path = { ...this.state.unit_by_path };
-                delete copy_unit_by_path[path];
-                this.setState({
-                    unit_by_path: copy_unit_by_path,
-                });
-            } else {
-                this.updateProperties(args[1], path);
+            // ignore uninteresting unit types
+            if (!path.endsWith("service") && !path.endsWith("timer") && !path.endsWith("socket") &&
+                !path.endsWith("target") && !path.endsWith("path"))
+                return;
+
+            if (iface === s_bus.I_TIMER) {
+                if (!this.timers[path])
+                    this.timers[path] = {};
+                this.addTimerProperties(props, this.timers[path]);
+                debug("timer PropertiesChanged on", path, JSON.stringify(this.timers[path]));
+                return;
+            }
+
+            // ignore uninteresting interfaces
+            if (iface !== s_bus.I_UNIT)
+                return;
+
+            let unit = this.units[path];
+
+            if (!unit) {
+                // this happens when starting an unloaded unit; unfortunately Units props is very incomplete, so we need a GetAll
+                debug("unit PropertiesChanged on previously unloaded unit", path);
+                try {
+                    [props] = await systemd_client[this.props.owner].call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_UNIT]);
+                } catch (ex) {
+                    console.warn("GetAll Unit for unknown unit", path, "failed:", ex.toString());
+                    return;
+                }
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230306-134233-5d848c5b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)